### PR TITLE
CLOUDP-120448 Telemetry - Track Events for Help (fix)

### PR DESCRIPTION
**Pull request by fmenezes**
## Proposed changes

Telemetry - Track Events for Help, fix invalid flags not being tracked

_Jira ticket:_ CLOUDP-120448

Closes #[issue number]

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongocli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongocli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: cmd/atlas/atlas.go [hunk 1]
```diff
@@ -134,7 +136,11 @@ func mongoCLIConfigFilePath() (configPath string, err error) {
 
 func main() {
 	cobra.EnableCommandSorting = false
-	cobra.OnInitialize(createConfigFromMongoCLIConfig, initConfig)
+
+	createConfigFromMongoCLIConfig()
+	loadConfig()
+
+	cobra.OnInitialize(initProfile)
```
Comment (tibulca) [inline cmd/atlas/atlas.go:143]: can you please provide some details about the change? the new code seems to do exactly what the previous one was doing, is this affecting setting up telemetry somehow?

Comment (fmenezes) [reply cmd/atlas/atlas.go:143]: well the key here is to load configuration outside of the `OnInitialize` block, `OnInitialize` is only triggered if all flags are valid, if you trigger command `atlas clusters ls --blah` for instance configuration would never be loaded, hence `telemetry_enabled` would always be `false`.

Comment (tibulca) [reply cmd/atlas/atlas.go:143]: thanks for clarifying

Comment (fmenezes) [reply cmd/atlas/atlas.go:143]: also setting the profile depends on `profile` flag, so that bit has to stay on `OnInitialize` block

**Review by tibulca — APPROVED** [review]

**Pull request merged**